Move filter adding MailPoet image resolution [MAILPOET-5749]

lysyjan opened the pull request:
## Description

_N/A_

## Code review notes

- From my understanding, moving the filter that adds MailPoet image size to the NewsletterEditor page shouldn't break anything.

## QA notes

1. Create a newsletter containing at least one image and Post with the featured image. All photos should have a width higher than 1320px.
2. Send a preview and check that all images in the email are with MailPoet resolution (width 1320px)
3. Send the newsletter and check again the image sizes.
4. Test it for all email types

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5749]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-5749]: https://mailpoet.atlassian.net/browse/MAILPOET-5749?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

lysyjan committed:
Move hook adding MailPoet image size to NewsletterEditor admin page

[MAILPOET-5749]

costasovo commented on dev/php82/Dockerfile:
```diff
@@ -37,6 +37,11 @@ RUN cat /tmp/xdebug.ini >> $XDEBUGINI_PATH
 RUN docker-php-ext-install pdo_mysql
 RUN docker-php-ext-install mysqli
+# imagick is needed for generating image thumbnails
+RUN apt-get update; \
```
Hmm, I see we still use PHP:8.2 RC image. I think we should switch to `wordpress:php8.2-apache`. WordPress image contains imagemagic. 

@lysyjan  Have you tried switching the image? Apart from the image name/tag, the docker file is very similar to the one for php8.1 that uses the WordPress image. I would suggest trying to switch the image instead of installing image magic as an extra step. If the switching image doesn't work or would lead to more changes we could go with this code and create an extra ticket for image update. What do you think?

lysyjan replied:
Great point @costasovo!
I was focused only on searching for a reason why thumbnails are not generated.

The switching docker image works. I will edit this commit and push an updated version.

costasovo reviewed (commented):
I would suggest trying to fix the imagemagick issue by switching the base image. Please see my comment.

lysyjan committed:
Change PHP base image for local environment

For better functionality of WordPress on local env, I changed the base image for PHP.
[MAILPOET-5749]

lysyjan commented:
@costasovo Thank you for your suggestion. I dropped the previous commit and added a new one.

costasovo reviewed (approved):
@lysyjan Good job! Thanks for updating the dev Dockerfile for PHP8.2. Until now, we have been using PHP 8.2 RC in our dev environment.

veljkho reviewed (approved):
Tested via Litmus for preview and sending. The resolution of 1320px [was set only for post's images](https://d.pr/i/8PuScI) and [not added images](https://d.pr/i/tAZuAM). Is this expected @lysyjan ? In the two screenshots you can see 1320px and 980px for the same image. The above was part of `Posts` block and post's featured image, but below image was just an image block.
